### PR TITLE
Modify the profile option to specify external compile definitions.

### PR DESCRIPTION
**Pull request by robertsipka**

### CODE REVIEW

File: jerry-core/profiles/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+
+### About profile files
+
+Specify compile definitions in profile files to use when compiling the `jerry-core` target.
+
+The default profile is ``es5.1`` which disables the ArrayBuffer built-in.
+
+### Using profiles with the build system
+
+You can add profile for the build system in two way:
```
Comment (zherczeg) [inline jerry-core/profiles/README.md]: You can specify the profile for the build system in the following ways:

File: jerry-core/profiles/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  * with absolute path
+  * with a name (in this case the given profile name is assumed in the `profiles` folder)
```
Comment (zherczeg) [inline jerry-core/profiles/README.md]: (this options selects `profiles/$(name).profile` file)

File: jerry-core/profiles/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+#### Restrictions
+
+The profile files can contain comments, but cannot have multi-line compile definitions, and any comma characters.
```
Comment (zherczeg) [inline jerry-core/profiles/README.md]: Only single line options are allowed in the profile file. Any line starting with hash-mark is ignored. Semicolon character is not allowed.

File: jerry-core/profiles/README.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+### Example usage:
+
+#### 1. Using the build script
+
+```
+# assuming you are in jerryscript folder
+./tools/build.py --profile=/path/to/profile/my_profile.profile
```
Comment (zherczeg) [inline jerry-core/profiles/README.md]: /absolute/path/to/my_profile.any_extension

File: jerry-core/profiles/README.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+```
+
+or
+
+```
+# assuming you are in jerryscript folder
+./tools/build.py --profile=minimal
+```
```
Comment (zherczeg) [inline jerry-core/profiles/README.md]: This command selects the profile/minimal.profile file.

File: jerry-core/profiles/es2015-subset.profile [hunk 6]
```diff
@@ -0,0 +1 @@
+# Intentionally empty
```
Comment (zherczeg) [inline jerry-core/profiles/es2015-subset.profile]: Currently an empty profile.

File: jerry-core/profiles/README.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+#### 2. Using only CMake build system
+
+Set the FEATURE_PROFILE compile definition with one of the following values:
```
Comment (zherczeg) [inline jerry-core/profiles/README.md]: Set FEATURE_PROFILE option to one of the following values:

**Comment by robertsipka** [comment]
@zherczeg : Thanks! I've modified this patch based on your suggestions.

**Comment by zherczeg** [comment]
LGTM

File: tools/build.py [hunk 8]
```diff
@@ -60,7 +63,7 @@ def devhelp(help):
     parser.add_argument('--linker-flag', metavar='OPT', action='append', default=[], help='add custom linker flag')
     parser.add_argument('--lto', metavar='X', choices=['ON', 'OFF'], default='ON', type=str.upper, help='enable link-time optimizations (%(choices)s; default: %(default)s)')
     parser.add_argument('--port-dir', metavar='DIR', action='store', default=DEFAULT_PORT_DIR, help='add port directory (default: %(default)s)')
-    parser.add_argument('--profile', metavar='PROFILE', choices=['es5.1', 'minimal', 'es2015-subset'], default='es5.1', type=str.lower, help='specify the profile (%(choices)s; default: %(default)s)')
+    parser.add_argument('--profile', metavar='FILE', action='store', default=DEFAULT_PROFILE, help='specify profile file (default: %(default)s)')
```
Comment (LaszloLango) [inline tools/build.py]: If the directory of profiles is hard coded, then we can set the 'choices' attribute like before.

Comment (LaszloLango) [inline tools/build.py]: I see.

**Pull request merged**